src/libtiled/gidmapper.cpp

```diff
@@ -186,16 +179,96 @@ QByteArray GidMapper::encodeLayerData(const TileLayer &tileLayer,
     return tileData.toBase64();
 }
 
+QByteArray GidMapper::encodeChunkData(const TileLayer &tileLayer,
```

Comment:
Rather than duplicating this whole function, please just add a `QRect` parameter. If the provided rect is empty it could probably default to `QRect(0, 0, tileLayer.width(), tileLayer.height())`.